llvm/lib/CodeGen/GlobalISel/CallLowering.cpp

```diff
@@ -866,11 +898,16 @@ bool CallLowering::handleAssignments(ValueHandler &Handler,
       }
     }
 
-    // Now that all pieces have been assigned, re-pack the register typed values
-    // into the original value typed registers.
-    if (Handler.isIncomingArgumentHandler() && OrigVT != LocVT) {
-      // Merge the split registers into the expected larger result vregs of
-      // the original call.
+    // In case of indirect parameter passing load the value referred to by
```

Comment:
nit: simplify `if` statements due to common conditions?
```
if (Handler.isIncomingArgumentHandler() && OrigVT != LocVT) {
  // In case of indirect parameter passing load the value referred to by the argument
  if (VA.getLocInfo() == CCValAssign::Indirect)
    Handler.assignValueToAddress(Args[i].OrigRegs[0], Args[i].Regs[0], OrigTy,
                                 MachinePointerInfo{}, VA);
  else
     // Now that all pieces have been assigned, re-pack the register typed values
     // into the original value typed registers.
     buildCopyFromRegs(MIRBuilder, Args[i].OrigRegs, Args[i].Regs, OrigTy,
                        LocTy, Args[i].Flags[0]);
}

Reply:
> Now that all pieces have been assigned, re-pack the register typed values into the original value typed registers.

Why don't we have to do this for the indirect case?

Reply:
You are correct. This logic could be moved up into the inner loop. This extra code is only needed if we have the value in separate registers and we want to merge the contents of these registers after the passing of them.

I will make this more readable by moving this logic to the inner loop.